test/utils.cpp

```diff
@@ -1643,6 +1643,35 @@ TEST_CASE("unescape_url() takes a percent-encoded string and returns the string
 
 }
 
+TEST_CASE("preserve_quotes() preserves single quotes and double quotes, if any",
+	"[utils]")
+{
+	REQUIRE(utils::preserve_quotes("") == "''");
+	REQUIRE(utils::preserve_quotes("sss") == "'sss'");
+
+	SECTION("Preserves double quotes") {
+		REQUIRE(utils::preserve_quotes("\"sss\"") == "'\"sss\"'");
+		REQUIRE(utils::preserve_quotes("sss\"") == "'sss\"'");
+		REQUIRE(utils::preserve_quotes("\"sss") == "'\"sss'");
+		REQUIRE(utils::preserve_quotes("ss\"s") == "'ss\"s'");
+	}
+
+	SECTION("Preserves single quotes") {
+		REQUIRE(utils::preserve_quotes("'sss'") == "\\\''sss'\\\'");
+		REQUIRE(utils::preserve_quotes("'sss") == "\\\''sss'");
+		REQUIRE(utils::preserve_quotes("sss'") == "'sss'\\\'");
+		REQUIRE(utils::preserve_quotes("ss's") == "'ss'\\\''s'");
+	}
+
+	SECTION("Preserves a combination of single and double quotes") {
+		REQUIRE(utils::preserve_quotes("'\"sss\"'") == "\\\''\"sss\"'\\\'");
+		REQUIRE(utils::preserve_quotes("\"'sss'\"") == "'\"'\\\''sss'\\\''\"'");
+		REQUIRE(utils::preserve_quotes("\"sss'") == "'\"sss'\\\'");
+		REQUIRE(utils::preserve_quotes("'sss\"") == "\\\''sss\"'");
+		REQUIRE(utils::preserve_quotes("ss\"'s") == "'ss\"'\\\''s'");
+	}
+}
```

Comment:
I wonder what happens if there are multiple single quotes in a row, like `it''s`. This isn't valid English punctuation, of course, but Newsboat shouldn't bork it either way.